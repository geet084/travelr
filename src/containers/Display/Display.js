import React, { Component } from 'react';
import { connect } from 'react-redux';
import { handleImages } from '../../thunks/handleImages';
import { getImageSuccess } from '../../actions/';
import PropTypes from 'prop-types';

export class Display extends Component {

  componentDidMount = () => {
    let { info, handleImages } = this.props;
    const url = `https://images-api.nasa.gov/asset/${info.images[0]}`
    
    if (info.images !== []) handleImages(url, getImageSuccess)
  }

  render() {
    const { info } = this.props;
    const { currentImage } = this.props.images;
    
    return (
      <div className='display'>
        {info && <p>{'Name  - ' + info.object_name}</p>}
        {info && <p>{'Avg. Temp  - ' + info.average_temp}</p>}
        {info && <p>{'Distance from the sun  - '}</p>}
        {info && <p>{info.distance_from_sun}</p>}
        {info && <p>{'Length of day  - ' + info.length_of_day}</p>}
        {info && <p>{'Orbital period  - ' + info.orbital_period}</p>}

        {currentImage.href !== '' && <img className='imgs' src={currentImage.href} alt="" />}
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  handleImages: (url, actionToDispatch) => dispatch(handleImages(url, actionToDispatch))
})

export const mapStateToProps = (state) => ({
  images: state.images,
})

Display.propTypes = {
  handleImages: PropTypes.func,
  images: PropTypes.object,
  info: PropTypes.object,
}

Display.defaultProps = {
  images: {},
  info: {},
}

export default connect(mapStateToProps, mapDispatchToProps)(Display);