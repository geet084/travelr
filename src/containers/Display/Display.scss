.display {
  color: white;
  margin-left: 200px;
  padding: 50px;
  font-size: 2rem;
}

.imgs {
  position: fixed;
  top: 0;
  left: 200px;
  margin: auto;
  height: 100%;
  z-index: -1;
}