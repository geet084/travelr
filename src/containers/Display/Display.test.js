import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';
import { mapStateToProps, mapDispatchToProps } from './Display'

describe('Display', () => {
  let wrapper;
  const mockInfo = ''
  const mockProps = {
    fetchImages: jest.fn(),
    info: mockInfo
  }

  beforeEach(() => {
    wrapper = shallow(<Display {...mockProps} />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mockArr = [];
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.fetchImages(mockArr)
    expect(mockDispatch).toHaveBeenCalled()
  })
  
  it('should mapStateToProps', () => {
    const mockState = {
      images: [{}],
      bodies: ['fake data'],
      planets: ['things and stuff'],
    }
    const expected = {
      images: [{}]
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  })
})