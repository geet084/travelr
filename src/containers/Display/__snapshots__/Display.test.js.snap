// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display initial state with NO props should match snapshot with no props 1`] = `
<div
  className="display"
>
  <p>
    Name  - undefined
  </p>
  <p>
    Avg. Temp  - undefined
  </p>
  <p>
    Distance from the sun  - 
  </p>
  <p>
    undefined / undefined
  </p>
  <p>
    Length of day  - undefined
  </p>
  <p>
    Orbital period  - undefined
  </p>
</div>
`;

exports[`Display initial state with props should match snapshot with props 1`] = `
<div
  className="display"
>
  <p>
    Name  - undefined
  </p>
  <p>
    Avg. Temp  - undefined
  </p>
  <p>
    Distance from the sun  - 
  </p>
  <p>
    undefined / undefined
  </p>
  <p>
    Length of day  - undefined
  </p>
  <p>
    Orbital period  - undefined
  </p>
  <img
    alt=""
    className="imgs"
    src="some href"
  />
</div>
`;
